fix: Package path resolution broken within @tamagui/vite-plugin on Windows platforms

CarlinJ216 opened the pull request:
Fix #3302

natew commented:
url package doesn't exist right? also i think most env has URL already natively?

CarlinJ216 commented:
> url package doesn't exist right? also i think most env has URL already natively?

`url` is the built-in node.js module-- (my understanding is that vite is a node.js script... that the vite config & plug-ins are ran within a node.js context.)

So, yep, this is the native URL being used.

CarlinJ216 commented:
> url package doesn't exist right? also i think most env has URL already natively?

Wait, I get what you are saying... While the `url` package does exist, no we don't need to explicitly import it, since `URL` is available as a node.js global. (I think VSCode automatically added the import...)

CarlinJ216 commented:
Updated the PR again...

Welp, your intuition that there must be a built-in way to solve this was spot-on...just not with Vite, but node itself! Haha. 

My first commits represented my narrow desire to simply swap out the use of String::replace in the existing logic (`const resolve = (name: string) => import.meta.resolve?.(name).replace('file://', '')`), with a "replacement" mechanism that was properly cross-platform. And since it is a fileURL we are dealing with, I figured using the node URL module (plus regex) would be safe & simple...

Lo and behold, I ended up re-inventing an existing built-in function! 🤦🏾‍♂️: [url.fileURLToPath](https://nodejs.org/docs/latest/api/url.html#urlfileurltopathurl-options) 

> import { fileURLToPath } from 'node:url';
> 
> const __filename = fileURLToPath(import.meta.url);
> 
> new URL('file:///C:/path/').pathname;      // Incorrect: /C:/path/
> fileURLToPath('file:///C:/path/');               // Correct:   C:\path\ (Windows)

I switched the implementation to simply calling it. However, because `import.meta.resolve?` can (possibly) eval to an `undefined` value, and this built-in `url.fileURLToPath` throws an Error on an "Invalid URL" input, we still need a simple existence check before passing the fileURL to it.

carlin-twd commented:
> Vite maybe has a built-in resolve for this

Also, I did search Vite's docs, the codebase, and a handful of other vite plugins that deal with paths. It does not appear that Vite exposes/exports any (path wrangling functions) that would help us here. All the other plugins seem to directly work with the built-in node `path` and `url` modules.

CarlinJ216 commented:
> Vite maybe has a built-in resolve for this

Also, I did search Vite's docs, the codebase, and a handful of other vite plugins that deal with paths. It does not appear that Vite exposes/exports any (path wrangling functions) that would help us here. All the other plugins seem to directly work with the built-in node `path` and `url` modules.

CarlinJ216 commented:
Is there an alternative approach that you would like me to explore? (More than willing to do so.) As mentioned on the linked Issue (#3302), the problem this PR addresses was introduced in the v1.112.2 release.
